### ISSUE TITLE
Get heart rate from other application

### ISSUE THREAD

**Issue by Almtesh** [Freeyourgadget/Gadgetbridge#435]
On Mi band 2, I saw I could get the heart rate in debug mode.
Would that be possible to get this from another app (i.e. Termux). I'd like to monitor this with a script.
Thanks.

**Comment by ashimokawa** [comment]
Do you really need a script or would it be okay if you would send intents (with a custom permission) so that you can write an Android app and consume these?

**Comment by Almtesh** [comment]
@ashimokawa OK, but I can't code Android apps. I only know Python and bash and I have some memories about C/C++ programming and that's all.

**Comment by cpfeiffer** [comment]
It's all a matter of willingness :-) This very project was created by people who have never written anything for Android before.

But we might also create a text file interface for command line tools.

**Comment by Almtesh** [comment]
What project do you talk about?

**Comment by ashimokawa** [comment]
He is talking about this project: Gadgetbridge.
We created it with very little or even zero Android programming knowledge. ;)

**Comment by ashimokawa** [comment]
He is talking about this project: Gadgetbridge.
We created it with very little or even zero Android programming knowledge. ;)

**Comment by Almtesh** [comment]
Do you mean that you learnt everything on Android programming to make Gadgetbridge or you used a tool to make Android apps without using Android programming?

**Comment by ashimokawa** [comment]
@Almtesh 
No I mean "learning by doing". We learned during the creation of Gadgetbridge

**Comment by Almtesh** [comment]
Yes, that, this is what I do, but I must first install a developpement environment and I don't know what I must install. I want my code to be compiled and tested, but my computer is not on Android.
What do you use for developpement environment?

**Comment by ashimokawa** [comment]
@Almtesh 
Core team uses Debian GNU/Linux + Android Studio 2.2
https://developer.android.com/studio/index.html
It is for available Windows and Mac also, but we never used that.
you can use any Android device connected to USB to run your code straight from Android studio on the device.

**Comment by Almtesh** [comment]
It took me several minutes to find how to launch the download, because clicking the links does nothing. So I downloaded through wget.

I can't find:
- the interface designer
- the compiled jar
- the virtual machine for test
- the module to send the app to my phone to test it
- the source code
- the documentation

For the moment, Android Studio is, for me, a text editor and template generator.
Is there any simple application to code on Android for people how don't understand java?

For memory, I openned this issue to see if I could get the heart rate simplelly.

**Comment by danielegobbetti** [comment]
> For memory, I openned this issue to see if I could get the heart rate simplelly.

@Almtesh I guess the answer is no. I changed the title of the issue to something more generic. I am not closing the issue as interacting with other apps is something that is within the scope of the project. Dealing with CS101 issues like installing apps and getting dev environments to work however is off topic.
